File: backend/app/models/pdf.py
"""
models/pdf.py — Pydantic schemas for file upload.

MODEL layer: pure data shapes, no logic.
Supports PDF, JPG, JPEG, PNG (the formats accepted by the existing app).
"""

from pydantic import BaseModel


class FileUploadResponse(BaseModel):
    """Returned immediately after a file is accepted."""

    status: str
    job_id: str
    filename: str
    size: int
    message: str


# Keep backwards-compatible alias
PDFUploadResponse = FileUploadResponse

File: frontend/src/models/pdf.ts
/**
 * models/pdf.ts — TypeScript mirror of backend FileUploadResponse.
 *
 * MODEL layer: pure data shape, no methods.
 * Keep in sync with backend/app/models/pdf.py :: FileUploadResponse.
 */

export interface FileUploadResponse {
  status: string
  job_id: string
  filename: string
  size: number
  message: string
}
